### PR TITLE
fix variable type

### PR DESCRIPTION
**Pull request by zloykaban**
#58

### CODE REVIEW

**Comment by jyaworski** [comment]
You may want to move the comparison up a line to before the size assignment. Also, would it be acceptable to do something like the following, with a guard clause:

```ruby
raise ArgumentError, "Size must be a positive integer: #{value.inspect}" unless value.is_a? Numeric
```

**Comment by zloykaban** [comment]
@jyaworski actualy seems that a good solution - remove test for float in input

File: lib/puppet/type/dhparam.rb [hunk 1]
```diff
@@ -18,7 +18,10 @@
     newvalues /\d+/
     defaultto 512
     validate do |value|
-      size = value
+      size = value.to_i
+      if value.is_a? Float
+        raise ArgumentError, "Size must be a positive integer: #{value.inspect}"
+      end
       if size <= 0 || size != size.to_i
```
Comment (raphink) [inline lib/puppet/type/dhparam.rb]: `size != size.to_i` doesn't make sense anymore.

**Comment by raphink** [comment]
For this kind of bug, I'd rather have a failing unit test reproducing #58 along with the patch fixing it.

**Comment by bandwith** [comment]
I have no idea how to write a unit test. I can confirm this bug on `puppet 3.8.1`.

**Comment by bandwith** [comment]
#64

**Comment by raphink** [comment]
Is this still necessary?

**Pull request closed without merging**